### PR TITLE
INSTUI-3089 - TS typing `ui-text-input`

### PR DESCRIPTION
**Pull request by ke1k0**

### CODE REVIEW

File: packages/ui-text-input/src/TextInput/__examples__/TextInput.examples.tsx [hunk 1]
```diff
@@ -69,16 +71,14 @@ export default {
...
-  filter: (props) => {
+  filter: (props: TextInputProps) => {
     return (
       (props.display === 'inline-block' && !props.width) ||
       (props.renderBeforeInput && props.width) ||
       (props.renderAfterInput && props.width) ||
       props.interaction === 'readonly' ||
       (props.placeholder && props.width) ||
-      (props.size !== 'medium' && props.width) ||
-      (props.size !== 'medium' && props.wrap)
```
Comment (ke1k0) [inline packages/ui-text-input/src/TextInput/__examples__/TextInput.examples.tsx:81]: The component doesn't have a 'size' prop

File: packages/ui-text-input/src/TextInput/index.tsx [hunk 2]
```diff
@@ -267,14 +251,12 @@ class TextInput extends Component<TextInputProps, TextInputState> {
...
         id={this.id}
         label={callRenderProp(renderLabel)}
-        // @ts-expect-error ts-migrate(2339) FIXME: Property '_messagesId' does not exist on type 'Tex... Remove this comment to see the full error message
         messagesId={this._messagesId}
         messages={messages}
         inline={display === 'inline-block'}
         width={width}
         inputContainerRef={inputContainerRef}
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'layout' does not exist on type 'Readonly... Remove this comment to see the full error message
-        layout={this.props.layout} // eslint-disable-line react/prop-types
```
Comment (ke1k0) [inline packages/ui-text-input/src/TextInput/index.tsx:277]: Added a note about this in the props.ts

File: packages/ui-text-input/src/TextInput/props.ts [hunk 3]
```diff
@@ -38,153 +38,186 @@ import type {
...
-   * provided via the `width` prop. Only applicable if `isInline={true}`.
+   * The width of the input (integer value 0 or higher), if a width is not explicitly
+   * provided via the `width` prop.
+   *
+   * Only applicable if `isInline={true}`.
+   *
+   * For more see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/size
    */
-  htmlSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  htmlSize?: number
```
Comment (ke1k0) [inline packages/ui-text-input/src/TextInput/props.ts:107]: This is a bugfix so I pushed it as a separate commit

File: packages/ui-text-input/src/TextInput/props.ts [hunk 4]
```diff
@@ -38,153 +38,186 @@ import type {
...
+  interaction?: InteractionType
+
   /**
-   * object with shape: `{
-   * text: PropTypes.node,
-   * type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
-   *   }`
+   * Array of objects with shape: `{
+   *   text: PropTypes.node,
+   *   type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
```
Comment (matyasf) [inline packages/ui-text-input/src/TextInput/props.ts:79]: I think this description could be more readable, something like:
Array of objects with shape: ```{
text: React.ReactNode,
type: One of  'error', 'hint', 'success', 'screenreader-only'
```

**Review by matyasf — APPROVED** [review]
looks good, just some small documentation fixes would be nice

**Review by Brailor — APPROVED** [review]

**Pull request merged**